### PR TITLE
Adding Location API methods

### PR DESCRIPTION
**Pull request by michaelaird**
Added 
createLocation
retrieveLocation
updateLocation

https://developer.squareup.com/reference/square/locations-api

### CODE REVIEW

File: types/square-connect/index.d.ts [hunk 1]
```diff
@@ -9191,6 +9191,18 @@ export class LocationsApi {
...
+   */
+    createLocation(body: Location): Promise<CreateLocationResponse>;
+  /**
+   * Retrieves details of a location.
+   */
+    retrieveLocation(locationId: string): Promise<RetrieveLocationResponse>;
+  /**
+   * Updates a location.
+   */
+    updateLocation(locationId: string, body: Location): Promise<UpdateLocationResponse>
```
Comment (mootrichard) [inline types/square-connect/index.d.ts:9205]: I believe this is erring because of a missing semi-colon.

**Comment by danger-public** [comment]
Inspecting the JavaScript source for this package found some properties that are not in the .d.ts files.
The check for missing properties isn't always right, so take this list as advice, not a requirement.
## square-connect ([<kbd>unpkg</kbd>](https://unpkg.com/browse/square-connect@latest/))
was missing the following properties:
1. CatalogV1Id
1. Currency
1. Error
1. GetBankAccountByV1IdResponse
1. OrderFulfillmentPickupDetailsScheduleType

<details>
<summary>as well as these 164 other properties...</summary>
<p>V1AdjustInventoryRequest, V1AdjustInventoryRequestAdjustmentType, V1ApplyFeeRequest, V1ApplyModifierListRequest, V1BankAccount, V1BankAccountType, V1CashDrawerEvent, V1CashDrawerEventEventType, V1CashDrawerShift, V1CashDrawerShiftEventType, V1Category, V1CreateCategoryRequest, V1CreateDiscountRequest, V1CreateEmployeeRoleRequest, V1CreateFeeRequest, V1CreateItemRequest, V1CreateModifierListRequest, V1CreateModifierOptionRequest, V1CreatePageRequest, V1CreateRefundRequest, V1CreateRefundRequestType, V1CreateVariationRequest, V1DeleteCategoryRequest, V1DeleteDiscountRequest, V1DeleteFeeRequest, V1DeleteItemRequest, V1DeleteModifierListRequest, V1DeleteModifierOptionRequest, V1DeletePageCellRequest, V1DeletePageRequest, V1DeleteTimecardRequest, V1DeleteTimecardResponse, V1DeleteVariationRequest, V1Discount, V1DiscountColor, V1DiscountDiscountType, V1Employee, V1EmployeeRole, V1EmployeeRolePermissions, V1EmployeeStatus, V1Fee, V1FeeAdjustmentType, V1FeeCalculationPhase, V1FeeInclusionType, V1FeeType, V1InventoryEntry, V1Item, V1ItemColor, V1ItemImage, V1ItemType, V1ItemVisibility, V1ListBankAccountsRequest, V1ListBankAccountsResponse, V1ListCashDrawerShiftsRequest, V1ListCashDrawerShiftsResponse, V1ListCategoriesRequest, V1ListCategoriesResponse, V1ListDiscountsRequest, V1ListDiscountsResponse, V1ListEmployeeRolesRequest, V1ListEmployeeRolesResponse, V1ListEmployeesRequest, V1ListEmployeesRequestStatus, V1ListEmployeesResponse, V1ListFeesRequest, V1ListFeesResponse, V1ListInventoryRequest, V1ListInventoryResponse, V1ListItemsRequest, V1ListItemsResponse, V1ListLocationsRequest, V1ListLocationsResponse, V1ListModifierListsRequest, V1ListModifierListsResponse, V1ListOrdersRequest, V1ListOrdersResponse, V1ListPagesRequest, V1ListPagesResponse, V1ListPaymentsRequest, V1ListPaymentsResponse, V1ListRefundsRequest, V1ListRefundsResponse, V1ListSettlementsRequest, V1ListSettlementsRequestStatus, V1ListSettlementsResponse, V1ListTimecardEventsRequest, V1ListTimecardEventsResponse, V1ListTimecardsRequest, V1ListTimecardsResponse, V1Merchant, V1MerchantAccountType, V1MerchantBusinessType, V1MerchantLocationDetails, V1ModifierList, V1ModifierListSelectionType, V1ModifierOption, V1Money, V1Order, V1OrderHistoryEntry, V1OrderHistoryEntryAction, V1OrderState, V1Page, V1PageCell, V1PageCellObjectType, V1PageCellPlaceholderType, V1Payment, V1PaymentDiscount, V1PaymentItemDetail, V1PaymentItemization, V1PaymentItemizationItemizationType, V1PaymentModifier, V1PaymentSurcharge, V1PaymentSurchargeType, V1PaymentTax, V1PaymentTaxInclusionType, V1PhoneNumber, V1Refund, V1RefundType, V1RemoveFeeRequest, V1RemoveModifierListRequest, V1RetrieveBankAccountRequest, V1RetrieveBusinessRequest, V1RetrieveCashDrawerShiftRequest, V1RetrieveEmployeeRequest, V1RetrieveEmployeeRoleRequest, V1RetrieveItemRequest, V1RetrieveModifierListRequest, V1RetrieveOrderRequest, V1RetrievePaymentRequest, V1RetrieveSettlementRequest, V1RetrieveTimecardRequest, V1Settlement, V1SettlementEntry, V1SettlementEntryType, V1SettlementStatus, V1Tender, V1TenderCardBrand, V1TenderEntryMethod, V1TenderType, V1Timecard, V1TimecardEvent, V1TimecardEventEventType, V1UpdateCategoryRequest, V1UpdateDiscountRequest, V1UpdateEmployeeRequest, V1UpdateEmployeeRoleRequest, V1UpdateFeeRequest, V1UpdateItemRequest, V1UpdateModifierListRequest, V1UpdateModifierListRequestSelectionType, V1UpdateModifierOptionRequest, V1UpdateOrderRequest, V1UpdateOrderRequestAction, V1UpdatePageCellRequest, V1UpdatePageRequest, V1UpdateTimecardRequest, V1UpdateVariationRequest, V1Variation, V1VariationInventoryAlertType, V1VariationPricingType, V1EmployeesApi, V1ItemsApi, V1LocationsApi, V1TransactionsApi</p>
</details>

<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against f99f7b11448d21c0b3dddcd8909b8ebf9a111bcc
</p>

**Review by mootrichard — APPROVED** [review]

**Comment by michaelaird** [comment]
@mootrichard thanks for reviewing.

I can't tell if this got merged and released or not?

**Comment by mootrichard** [comment]
@michaelaird Yeah, I'm not sure it has been. I don't really understand the new process for DefinitelyTyped, I guess it needs to get merged by a core member of the project? 

I definitely approved it. 🤷‍♂️

**Review by ddimitrioglo — APPROVED** [review]

**Comment by ddimitrioglo** [comment]
@danger-public (or it's owner) if you want to use API v1 endpoints - just use @types/square-connect v2 ([example](https://www.npmjs.com/package/@types/square-connect/v/2.20190814.3))

**Comment by michaelaird** [comment]
Ready to merge

**Pull request merged**